common/src/io.cpp

```diff
@@ -184,7 +186,8 @@ pcl::concatenateFields (const pcl::PCLPointCloud2 &cloud1,
   int point_offset = 0;
   for (std::size_t cp = 0; cp < cloud_out.width * cloud_out.height; ++cp)
   {
-    memcpy (&cloud_out.data[point_offset], &cloud2.data[cp * cloud2.point_step], cloud2.point_step);
+    std::copy_n(&cloud2.data[cp * cloud2.point_step], cloud2.point_step,
```

Comment:
In some places, you've used `cloud.data.data() + index` while in other places, you've used `&cloud.data[index]`. Maybe choose one (I'd say the latter since it's shorter) when `begin` isn't available.